### ISSUE TITLE
[Appsmith] runaway or idle connections 

### ISSUE THREAD

**Issue by rohan-arthur** [supabase/supabase#17549]
# Bug report

- [x ] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Coming here from Appsmith, and we are noticing a high number of connections (idle and otherwise) when connecting with supabase.
Assumption: Table `pg_stat_activity` lists the number of connections to the db, and column `application_name` contains the caller or client.

The consequence is that within appsmith, we have a configurable limit beyond which we just show a "timeout" error - we should improve this of course, but:

As you can see in the attached screenshot, there's a very large number of connections opened from supabase. This is the reason for reporting a bug here, as there may be something for supabase to look into. Happy to discuss this further!

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Please see the attached screenshot where:

1. I have connections to supabase and neon
2. In each db, I have a dummy table
3. For each, I run a select on the dummy table followed by a count of number of connections with application_name as "Appsmith JDBC Driver"

##Findings
I started this experiment in the morning around 1100, and it is now 1300. 
1. At the time of starting, both connections were 3
2. In the case of Supabase, the count kept increasing. In the other case the count increased, but also quickly kept decreasing after some inactivity
3. Currently, the count is 50 for supabase and 4 for neon

## Expected behavior

A reasonable number of connections, but we see incrementing large numbers of them.

## Screenshots
![Screenshot 2023-09-19 at 12 59 24](https://github.com/supabase/supabase/assets/94514895/f7761a4c-2aed-4bf4-b0cc-78cb11ef439b)

## System information

- OS: macOS
- Browser chrome
- Version of supabase-js: cloud
- Version of Node.js: na

**Comment by chasers** [comment]
Can you confirm the connection strings of each service? 

With Supabase I assume you're connecting directly to Postgres? Using port 5432?

You may be using a pooler with one and connecting directly with the other.

**Comment by rohan-arthur** [comment]
I just used the connection params like in the screenshot
![Screenshot 2023-09-20 at 11 01 04](https://github.com/supabase/supabase/assets/94514895/5576bb28-d801-4dbf-b69e-8b667ca2841a)

**Comment by encima** [comment]
Hi  @rohan-arthur 
Is this still occurring with Appsmith? If you would like, we can share a project in order to debug this further?

**Comment by rohan-arthur** [comment]
Here's an app that demonstrates the problem: https://app.appsmith.com/app/idle-connections/supabase-63fddd6b14130d22285c0057

1. click `select supabase` to trigger a `select * from dummy_table` query
2. click `supabase connections` to see the number of open connections

do the same for neon if you want a comparison.

**Comment by encima** [comment]
Hi @rohan-arthur 
Sorry for the delay in getting to this, that is on me. I assume your Supabase project is free so it may have been paused? If you can let me know the project_ref then I can make sure it is not paused while we investigate this

**Comment by encima** [comment]
Hi @rohan-arthur 
I attempted to check your instance but your app fails with "failed to fetch" so unsure if there is an issue there?

I am able to reproduce your issue, though. Here is the sample [app](https://app.appsmith.com/app/supasmith-connections/direct-conns-65d320714f902f01cc5b6da8).

I have added some pages that do the same but uses the Supavisor connection pooler 
(configured for transaction and session) and this does show connections being terminated instantly (although there is no delay in place there).

A non Supabase DB was used (the last page) which also shows an increasing conn count (albeit 4-5x less)
